File: models/meal_plan.py
from . import db
from datetime import datetime

class MealPlan(db.Model):
    __tablename__ = 'meal_plans'
    
    plan_id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.user_id'), nullable=False)
    week_start = db.Column(db.Date, nullable=False)
    week_end = db.Column(db.Date, nullable=False)
    title = db.Column(db.String(255), default='Meal Plan')
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    items = db.relationship('MealPlanItem', backref='meal_plan', lazy='dynamic')
    
    def __repr__(self):
        return f'<MealPlan {self.title}>'


class MealPlanItem(db.Model):
    __tablename__ = 'meal_plan_items'
    
    item_id = db.Column(db.Integer, primary_key=True)
    plan_id = db.Column(db.Integer, db.ForeignKey('meal_plans.plan_id'), nullable=False)
    recipe_id = db.Column(db.Integer, db.ForeignKey('recipes.recipe_id'), nullable=False)
    meal_date = db.Column(db.Date, nullable=False)
    meal_type = db.Column(db.Enum('breakfast', 'lunch', 'dinner', 'snack'), nullable=False)
    is_cooked = db.Column(db.Boolean, default=False)
    cooked_at = db.Column(db.DateTime)
    notes = db.Column(db.Text)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    def __repr__(self):
        return f'<MealPlanItem {self.meal_type} - {self.meal_date}>'

File: dotnet_backend/BepTroLy.API/Models/MealPlan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BepTroLy.API.Models;

[Table("meal_plans")]
public class MealPlan
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("plan_id")]
    public int PlanId { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [Required]
    [Column("week_start")]
    public DateOnly WeekStart { get; set; }

    [Required]
    [Column("week_end")]
    public DateOnly WeekEnd { get; set; }

    [MaxLength(255)]
    [Column("title")]
    public string Title { get; set; } = "Meal Plan";

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey("UserId")]
    public User? User { get; set; }

    public ICollection<MealPlanItem> Items { get; set; } = new List<MealPlanItem>();
}

[Table("meal_plan_items")]
public class MealPlanItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("item_id")]
    public int ItemId { get; set; }

    [Column("plan_id")]
    public int PlanId { get; set; }

    [Column("recipe_id")]
    public int RecipeId { get; set; }

    [Required]
    [Column("meal_date")]
    public DateOnly MealDate { get; set; }

    // Enum: 'breakfast', 'lunch', 'dinner', 'snack'
    [Required]
    [Column("meal_type")]
    public string MealType { get; set; } = string.Empty;

    [Column("is_cooked")]
    public bool IsCooked { get; set; } = false;

    [Column("cooked_at")]
    public DateTime? CookedAt { get; set; }

    [Column("notes", TypeName = "text")]
    public string? Notes { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey("PlanId")]
    public MealPlan? MealPlan { get; set; }

    [ForeignKey("RecipeId")]
    public Recipe? Recipe { get; set; }
}
